--- a/setup.py.orig	2020-07-18 22:22:56.917942280 -0400
+++ b/setup.py	2020-07-18 22:37:39.203899069 -0400
@@ -26,7 +26,6 @@
     url="https://github.com/SymbiFlow/python-sdf-timing",
     packages=setuptools.find_packages(),
     install_requires=['ply', 'pyjson'],
-    setup_requires=["pytest-runner"],
     tests_require=["pytest"],
     classifiers=[
         "Programming Language :: Python :: 3",
@@ -35,4 +36,4 @@
     entry_points={
         'console_scripts': ['sdf_timing_parse=sdf_timing.sdfparse:main'],
     },
-)
+)
\ No newline at end of file
