diff --git a/faiss/python/CMakeLists.txt b/faiss/python/CMakeLists.txt
index 3fc46f4c5..02f24897d 100644
--- a/faiss/python/CMakeLists.txt
+++ b/faiss/python/CMakeLists.txt
@@ -189,12 +189,12 @@ set_property(TARGET faiss_example_external_module PROPERTY SWIG_COMPILE_OPTIONS
 
 if(NOT WIN32)
   # NOTE: Python does not recognize the dylib extension.
-  set_target_properties(swigfaiss PROPERTIES SUFFIX .so)
-  set_target_properties(swigfaiss_avx2 PROPERTIES SUFFIX .so)
-  set_target_properties(swigfaiss_avx512 PROPERTIES SUFFIX .so)
-  set_target_properties(swigfaiss_avx512_spr PROPERTIES SUFFIX .so)
-  set_target_properties(swigfaiss_sve PROPERTIES SUFFIX .so)
-  set_target_properties(faiss_example_external_module PROPERTIES SUFFIX .so)
+  set_target_properties(swigfaiss PROPERTIES SUFFIX .so INSTALL_RPATH "$ORIGIN/." BUILD_WITH_INSTALL_RPATH TRUE)
+  set_target_properties(swigfaiss_avx2 PROPERTIES SUFFIX .so INSTALL_RPATH "$ORIGIN/." BUILD_WITH_INSTALL_RPATH TRUE)
+  set_target_properties(swigfaiss_avx512 PROPERTIES SUFFIX .so INSTALL_RPATH "$ORIGIN/." BUILD_WITH_INSTALL_RPATH TRUE)
+  set_target_properties(swigfaiss_avx512_spr PROPERTIES SUFFIX .so INSTALL_RPATH "$ORIGIN/." BUILD_WITH_INSTALL_RPATH TRUE)
+  set_target_properties(swigfaiss_sve PROPERTIES SUFFIX .so INSTALL_RPATH "$ORIGIN/." BUILD_WITH_INSTALL_RPATH TRUE)
+  set_target_properties(faiss_example_external_module PROPERTIES SUFFIX .so INSTALL_RPATH "$ORIGIN/." BUILD_WITH_INSTALL_RPATH TRUE)
 else()
   # we need bigobj for the swig wrapper
   target_compile_options(swigfaiss PRIVATE /bigobj)
@@ -280,7 +280,7 @@ target_include_directories(swigfaiss_sve PRIVATE ${PROJECT_SOURCE_DIR}/../..)
 target_include_directories(faiss_example_external_module PRIVATE ${PROJECT_SOURCE_DIR}/../..)
 
 find_package(Python REQUIRED
-  COMPONENTS Development NumPy
+  COMPONENTS Development.Module NumPy
 )
 
 add_library(faiss_python_callbacks EXCLUDE_FROM_ALL
