diff --git a/faiss/python/CMakeLists.txt b/faiss/python/CMakeLists.txt
index 3fc46f4c5..02f24897d 100644
--- a/faiss/python/CMakeLists.txt
+++ b/faiss/python/CMakeLists.txt
@@ -189,11 +189,11 @@ set_property(TARGET faiss_example_external_module PROPERTY SWIG_COMPILE_OPTIONS
 
 if(NOT WIN32)
   # NOTE: Python does not recognize the dylib extension.
-  set_target_properties(swigfaiss PROPERTIES SUFFIX .so)
-  set_target_properties(swigfaiss_avx2 PROPERTIES SUFFIX .so)
-  set_target_properties(swigfaiss_avx512 PROPERTIES SUFFIX .so)
-  set_target_properties(swigfaiss_avx512_spr PROPERTIES SUFFIX .so)
-  set_target_properties(swigfaiss_sve PROPERTIES SUFFIX .so)
-  set_target_properties(faiss_example_external_module PROPERTIES SUFFIX .so)
+  set_target_properties(swigfaiss PROPERTIES SUFFIX .so INSTALL_RPATH "$ORIGIN/." BUILD_WITH_INSTALL_RPATH TRUE)
+  set_target_properties(swigfaiss_avx2 PROPERTIES SUFFIX .so INSTALL_RPATH "$ORIGIN/." BUILD_WITH_INSTALL_RPATH TRUE)
+  set_target_properties(swigfaiss_avx512 PROPERTIES SUFFIX .so INSTALL_RPATH "$ORIGIN/." BUILD_WITH_INSTALL_RPATH TRUE)
+  set_target_properties(swigfaiss_avx512_spr PROPERTIES SUFFIX .so INSTALL_RPATH "$ORIGIN/." BUILD_WITH_INSTALL_RPATH TRUE)
+  set_target_properties(swigfaiss_sve PROPERTIES SUFFIX .so INSTALL_RPATH "$ORIGIN/." BUILD_WITH_INSTALL_RPATH TRUE)
+  set_target_properties(faiss_example_external_module PROPERTIES SUFFIX .so INSTALL_RPATH "$ORIGIN/." BUILD_WITH_INSTALL_RPATH TRUE)
 else()
   # we need bigobj for the swig wrapper
   target_compile_options(swigfaiss PRIVATE /bigobj)