diff --git a/faiss/python/setup.py b/faiss/python/setup.py
index 8d2150e7..4876fb49 100644
--- a/faiss/python/setup.py
+++ b/faiss/python/setup.py
@@ -27,11 +27,13 @@ swigfaiss_generic_lib = f"{prefix}_swigfaiss{ext}"
 swigfaiss_avx2_lib = f"{prefix}_swigfaiss_avx2{ext}"
 swigfaiss_avx512_lib = f"{prefix}_swigfaiss_avx512{ext}"
 swigfaiss_sve_lib = f"{prefix}_swigfaiss_sve{ext}"
+callbacks_lib = f"{prefix}libfaiss_python_callbacks{ext}"
 
 found_swigfaiss_generic = os.path.exists(swigfaiss_generic_lib)
 found_swigfaiss_avx2 = os.path.exists(swigfaiss_avx2_lib)
 found_swigfaiss_avx512 = os.path.exists(swigfaiss_avx512_lib)
 found_swigfaiss_sve = os.path.exists(swigfaiss_sve_lib)
+found_callbacks = os.path.exists(callbacks_lib)
 
 assert (found_swigfaiss_generic or found_swigfaiss_avx2 or found_swigfaiss_avx512 or found_swigfaiss_sve), \
     f"Could not find {swigfaiss_generic_lib} or " \
@@ -58,6 +60,10 @@ if found_swigfaiss_sve:
     shutil.copyfile("swigfaiss_sve.py", "faiss/swigfaiss_sve.py")
     shutil.copyfile(swigfaiss_sve_lib, f"faiss/_swigfaiss_sve{ext}")
 
+if found_callbacks:
+    print(f"Copying {callbacks_lib}")
+    shutil.copyfile(callbacks_lib, f"faiss/{callbacks_lib}")
+
 long_description="""
 Faiss is a library for efficient similarity search and clustering of dense
 vectors. It contains algorithms that search in sets of vectors of any size,
