diff --git a/rpcs3/Emu/Audio/audio_resampler.h b/rpcs3/Emu/Audio/audio_resampler.h
index 56630df74..969b4df3e 100644
--- a/rpcs3/Emu/Audio/audio_resampler.h
+++ b/rpcs3/Emu/Audio/audio_resampler.h
@@ -8,7 +8,7 @@
 #pragma GCC diagnostic ignored "-Wsuggest-override"
 #pragma GCC diagnostic ignored "-Wold-style-cast"
 #endif
-#include "SoundTouch.h"
+#include "soundtouch/SoundTouch.h"
 #ifndef _MSC_VER
 #pragma GCC diagnostic pop
 #endif
diff --git a/rpcs3/Emu/Io/RB3MidiGuitar.h b/rpcs3/Emu/Io/RB3MidiGuitar.h
index 3a93f95d6..1eef08592 100644
--- a/rpcs3/Emu/Io/RB3MidiGuitar.h
+++ b/rpcs3/Emu/Io/RB3MidiGuitar.h
@@ -2,7 +2,7 @@
 
 #include "Emu/Io/usb_device.h"
 
-#include <rtmidi_c.h>
+#include <rtmidi/rtmidi_c.h>
 
 class usb_device_rb3_midi_guitar : public usb_device_emulated
 {
diff --git a/rpcs3/Emu/Io/RB3MidiKeyboard.h b/rpcs3/Emu/Io/RB3MidiKeyboard.h
index c5042af23..de6511bf3 100644
--- a/rpcs3/Emu/Io/RB3MidiKeyboard.h
+++ b/rpcs3/Emu/Io/RB3MidiKeyboard.h
@@ -2,7 +2,7 @@
 
 #include "Emu/Io/usb_device.h"
 
-#include <rtmidi_c.h>
+#include <rtmidi/rtmidi_c.h>
 
 class usb_device_rb3_midi_keyboard : public usb_device_emulated
 {
diff --git a/rpcs3/Emu/NP/upnp_handler.cpp b/rpcs3/Emu/NP/upnp_handler.cpp
index dca00de98..b30bb2e81 100644
--- a/rpcs3/Emu/NP/upnp_handler.cpp
+++ b/rpcs3/Emu/NP/upnp_handler.cpp
@@ -2,8 +2,8 @@
 #include "upnp_handler.h"
 #include "util/logs.hpp"
 
-#include <miniwget.h>
-#include <upnpcommands.h>
+#include <miniupnpc/miniwget.h>
+#include <miniupnpc/upnpcommands.h>
 
 LOG_CHANNEL(upnp_log, "UPNP");
 
diff --git a/rpcs3/Emu/NP/upnp_handler.h b/rpcs3/Emu/NP/upnp_handler.h
index 40892f744..bb99b88ae 100644
--- a/rpcs3/Emu/NP/upnp_handler.h
+++ b/rpcs3/Emu/NP/upnp_handler.h
@@ -2,7 +2,7 @@
 
 #include <unordered_map>
 
-#include <miniupnpc.h>
+#include <miniupnpc/miniupnpc.h>
 
 #include "upnp_config.h"
 #include "Utilities/mutex.h"
diff --git a/rpcs3/Emu/RSX/VK/VKCommonDecompiler.cpp b/rpcs3/Emu/RSX/VK/VKCommonDecompiler.cpp
index a9005d520..b241a7897 100644
--- a/rpcs3/Emu/RSX/VK/VKCommonDecompiler.cpp
+++ b/rpcs3/Emu/RSX/VK/VKCommonDecompiler.cpp
@@ -13,7 +13,7 @@
 #pragma clang diagnostic ignored "-Winconsistent-missing-override"
 #endif
 #endif
-#include "SPIRV/GlslangToSpv.h"
+#include "glslang/SPIRV/GlslangToSpv.h"
 #ifdef _MSC_VER
 #pragma warning(pop)
 #else
diff --git a/rpcs3/Input/ds4_pad_handler.cpp b/rpcs3/Input/ds4_pad_handler.cpp
index a7a484920..e30e225af 100644
--- a/rpcs3/Input/ds4_pad_handler.cpp
+++ b/rpcs3/Input/ds4_pad_handler.cpp
@@ -626,7 +626,7 @@ int ds4_pad_handler::send_output_report(DS4Device* device)
 		outputBuf[76] = (crcCalc >> 16) & 0xFF;
 		outputBuf[77] = (crcCalc >> 24) & 0xFF;
 
-		return hid_write_control(device->hidDevice, outputBuf.data(), DS4_OUTPUT_REPORT_0x11_SIZE);
+		return hid_write(device->hidDevice, outputBuf.data(), DS4_OUTPUT_REPORT_0x11_SIZE);
 	}
 	else
 	{
diff --git a/rpcs3/Input/hid_pad_handler.h b/rpcs3/Input/hid_pad_handler.h
index 943b6c243..69b80964d 100644
--- a/rpcs3/Input/hid_pad_handler.h
+++ b/rpcs3/Input/hid_pad_handler.h
@@ -4,7 +4,7 @@
 #include "Utilities/CRC.h"
 #include "Utilities/Thread.h"
 
-#include "hidapi.h"
+#include "hidapi/hidapi.h"
 
 #include <algorithm>
 
diff --git a/rpcs3/rpcs3qt/midi_creator.cpp b/rpcs3/rpcs3qt/midi_creator.cpp
index ff83461fe..165ab7dd9 100644
--- a/rpcs3/rpcs3qt/midi_creator.cpp
+++ b/rpcs3/rpcs3qt/midi_creator.cpp
@@ -4,7 +4,7 @@
 #include "Utilities/StrFmt.h"
 #include "Utilities/StrUtil.h"
 
-#include <rtmidi_c.h>
+#include <rtmidi/rtmidi_c.h>
 
 LOG_CHANNEL(cfg_log, "CFG");
 
