diff --git a/3rdparty/CMakeLists.txt b/3rdparty/CMakeLists.txt
index 4006a4920..e21b06242 100644
--- a/3rdparty/CMakeLists.txt
+++ b/3rdparty/CMakeLists.txt
@@ -78,13 +78,12 @@ endif()
 
 
 # hidapi
-add_subdirectory(hidapi)
-
+find_package(hidapi REQUIRED)
+add_library(hidapi INTERFACE)
 
 # Vulkan
-add_subdirectory(glslang EXCLUDE_FROM_ALL)
-#add_subdirectory(SPIRV EXCLUDE_FROM_ALL)
-
+find_package(glslang REQUIRED)
+add_library(glslang INTERFACE)
 
 # yaml-cpp
 add_subdirectory(yaml-cpp)
@@ -104,7 +103,6 @@ else()
 endif()
 
 # OpenGL
-
 find_package(OpenGL REQUIRED)
 
 add_library(3rdparty_opengl INTERFACE)
@@ -127,15 +125,20 @@ endif()
 add_library(3rdparty_stblib INTERFACE)
 target_include_directories(3rdparty_stblib INTERFACE stblib/include)
 
-
 # DiscordRPC
 add_subdirectory(discord-rpc)
 
 # Cubeb
-add_subdirectory(cubeb EXCLUDE_FROM_ALL)
+find_package(cubeb REQUIRED)
+add_library(cubeb INTERFACE)
+target_include_directories(cubeb INTERFACE)
+target_link_libraries(cubeb INTERFACE)
 
 # SoundTouch
-add_subdirectory(SoundTouch EXCLUDE_FROM_ALL)
+find_package(SoundTouch REQUIRED)
+add_library(SoundTouch INTERFACE)
+target_include_directories(SoundTouch INTERFACE)
+target_link_libraries(SoundTouch INTERFACE)
 
 # libevdev
 set(LIBEVDEV_TARGET 3rdparty_dummy_lib)
@@ -210,10 +213,16 @@ if(USE_VULKAN)
 endif()
 
 # AsmJit
-add_subdirectory(asmjit EXCLUDE_FROM_ALL)
+find_package(asmjit REQUIRED)
+add_library(asmjit INTERFACE)
+target_include_directories(asmjit INTERFACE)
+target_link_libraries(asmjit INTERFACE)
 
 # OpenAL
-add_subdirectory(OpenAL EXCLUDE_FROM_ALL)
+find_package(OpenAL REQUIRED)
+add_library(OpenAL INTERFACE)
+target_include_directories(OpenAL INTERFACE)
+target_link_libraries(OpenAL INTERFACE)
 
 # FAudio
 set(FAUDIO_TARGET 3rdparty_dummy_lib)
@@ -309,7 +318,6 @@ if(NOT MSVC)
 	target_link_libraries(3rdparty_glew INTERFACE GLEW::GLEW)
 endif()
 
-
 # LLVM
 add_subdirectory(llvm EXCLUDE_FROM_ALL)
 
@@ -344,10 +352,17 @@ if(USE_SDL)
 endif()
 
 # MINIUPNP
-add_subdirectory(miniupnp EXCLUDE_FROM_ALL)
+find_package(PkgConfig)
+pkg_check_modules(miniupnpc REQUIRED miniupnpc)
+add_library(miniupnpc INTERFACE)
+target_include_directories(miniupnpc INTERFACE)
+target_link_libraries(miniupnpc INTERFACE)
 
 # RTMIDI
-add_subdirectory(rtmidi EXCLUDE_FROM_ALL)
+find_package(rtmidi REQUIRED)
+add_library(rtmidi INTERFACE)
+target_include_directories(rtmidi INTERFACE)
+target_link_libraries(rtmidi INTERFACE)
 
 # add nice ALIAS targets for ease of use
 if(USE_SYSTEM_LIBUSB)
@@ -361,7 +376,7 @@ add_library(3rdparty::flatbuffers ALIAS 3rdparty_flatbuffers)
 add_library(3rdparty::pugixml ALIAS pugixml)
 add_library(3rdparty::yaml-cpp ALIAS yaml-cpp)
 add_library(3rdparty::xxhash ALIAS xxhash)
-add_library(3rdparty::hidapi ALIAS 3rdparty_hidapi)
+add_library(3rdparty::hidapi ALIAS hidapi)
 add_library(3rdparty::libpng ALIAS ${LIBPNG_TARGET})
 add_library(3rdparty::opengl ALIAS 3rdparty_opengl)
 add_library(3rdparty::stblib ALIAS 3rdparty_stblib)
@@ -369,12 +384,14 @@ add_library(3rdparty::discordRPC ALIAS 3rdparty_discordRPC)
 add_library(3rdparty::faudio ALIAS ${FAUDIO_TARGET})
 add_library(3rdparty::libevdev ALIAS ${LIBEVDEV_TARGET})
 add_library(3rdparty::vulkan ALIAS ${VULKAN_TARGET})
-add_library(3rdparty::openal ALIAS 3rdparty_openal)
+add_library(3rdparty::openal ALIAS OpenAL)
 add_library(3rdparty::ffmpeg ALIAS 3rdparty_ffmpeg)
 add_library(3rdparty::glew ALIAS 3rdparty_glew)
 add_library(3rdparty::wolfssl ALIAS wolfssl)
 add_library(3rdparty::libcurl ALIAS 3rdparty_libcurl)
-add_library(3rdparty::soundtouch ALIAS soundtouch)
+add_library(3rdparty::soundtouch ALIAS SoundTouch)
 add_library(3rdparty::sdl2 ALIAS ${SDL2_TARGET})
-add_library(3rdparty::miniupnpc ALIAS libminiupnpc-static)
+add_library(3rdparty::miniupnpc ALIAS miniupnpc)
+add_library(3rdparty::asmjit ALIAS asmjit)
 add_library(3rdparty::rtmidi ALIAS rtmidi)
+add_library(3rdparty::cubeb ALIAS cubeb)
