diff --git a/mupen64plus-qt.pro b/mupen64plus-qt.pro
index 7be60cc..3cef254 100644
--- a/mupen64plus-qt.pro
+++ b/mupen64plus-qt.pro
@@ -86,7 +86,8 @@ win32|macx|linux_quazip_static {
                 LIBS += -lquazip5
             }
         } else {
-            LIBS += -lquazip5
+            LIBS += -lquazip1-qt5
+            INCLUDEPATH += /usr/include/QuaZip
         }
     }
 }
diff --git a/src/common.cpp b/src/common.cpp
index 3ffdb7c..a9615ab 100644
--- a/src/common.cpp
+++ b/src/common.cpp
@@ -40,8 +40,8 @@
 #include <QLocale>
 #include <QSize>
 
-#include <quazip5/quazip.h>
-#include <quazip5/quazipfile.h>
+#include <quazip/quazip.h>
+#include <quazip/quazipfile.h>
 
 #ifdef Q_OS_WIN
 #include <QCoreApplication>
diff --git a/src/emulation/emulatorhandler.cpp b/src/emulation/emulatorhandler.cpp
index 2f89307..db2296a 100644
--- a/src/emulation/emulatorhandler.cpp
+++ b/src/emulation/emulatorhandler.cpp
@@ -38,8 +38,8 @@
 #include <QFile>
 #include <QMessageBox>
 
-#include <quazip5/quazip.h>
-#include <quazip5/quazipfile.h>
+#include <quazip/quazip.h>
+#include <quazip/quazipfile.h>
 
 
 EmulatorHandler::EmulatorHandler(QWidget *parent) : QObject(parent)
