diff --git a/pyproject.toml b/pyproject.toml
index d89ac45..30245fe 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -55,7 +55,7 @@ classifiers = [
 ]
 
 [tool.setuptools]
-# packages = ["cma", "cma.utilities"]  # fails when they are in an src/ folder
+packages = ["cma", "cma.utilities"]  # fails when they are in an src/ folder
 include-package-data = false  # true is default, false doesn't change anything
 
 # [tool.setuptools.packages.find]  # for . works only in a clean folder
