--- a/WORKSPACE	2020-09-26 23:37:56.849050723 -0400
+++ b/WORKSPACE	2020-09-26 23:38:42.215737175 -0400
@@ -12,6 +12,16 @@
     ],
 )
 
+http_archive(
+    name = "io_bazel_rules_docker",
+    sha256 = "4521794f0fba2e20f3bf15846ab5e01d5332e587e9ce81629c7f96c793bb7036",
+    strip_prefix = "rules_docker-0.14.4",
+    urls = [
+        "https://github.com/bazelbuild/rules_docker/archive/v0.14.4.tar.gz",
+    ],
+)
+
+
 # Load tf_repositories() before loading dependencies for other repository so
 # that dependencies like com_google_protobuf won't be overridden.
 load("//tensorflow:workspace.bzl", "tf_repositories")
