diff --git a/src/event/SignalMonitor.cxx b/src/event/SignalMonitor.cxx
index 8e0d3238e..07352d3a2 100644
--- a/src/event/SignalMonitor.cxx
+++ b/src/event/SignalMonitor.cxx
@@ -57,7 +57,7 @@ public:
 	explicit SignalMonitor(EventLoop &_loop)
 		:event(_loop, BIND_THIS_METHOD(OnSocketReady)) {
 #ifndef USE_SIGNALFD
-		event.Open(SocketDescriptor(fd.Get()));
+		event.Open(SocketDescriptor(fd.GetSocket()));
 		event.ScheduleRead();
 #endif
 	}
