diff --git a/CMakeLists.txt b/CMakeLists.txt
index e44a3ee8..d564a4e8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -245,13 +245,7 @@ if( NOT CMAKE_BUILD_TYPE )
     set(CMAKE_BUILD_TYPE Release)
 endif()
 
-if(CMAKE_CROSSCOMPILING)
-    set(BBA_IMPORT "IMPORTFILE-NOTFOUND" CACHE FILEPATH
-        "Path to the `bba-export.cmake` export file from a native build")
-    include(${BBA_IMPORT})
-else()
-    add_subdirectory(bba)
-endif()
+add_subdirectory(bba)
 
 include(TestBigEndian)
 test_big_endian(IS_BIG_ENDIAN)
diff --git a/xilinx/arch_place.cc b/xilinx/arch_place.cc
index b29e1243..0cd9819f 100644
--- a/xilinx/arch_place.cc
+++ b/xilinx/arch_place.cc
@@ -19,6 +19,7 @@
 
 #include <boost/algorithm/string.hpp>
 #include <queue>
+#include <set>
 #include "design_utils.h"
 #include "log.h"
 #include "nextpnr.h"
diff --git a/xilinx/fasm.cc b/xilinx/fasm.cc
index b42f8716..cd471d33 100644
--- a/xilinx/fasm.cc
+++ b/xilinx/fasm.cc
@@ -21,6 +21,7 @@
 #include <boost/algorithm/string.hpp>
 #include <boost/range/adaptor/reversed.hpp>
 #include <fstream>
+#include <set>
 #include "log.h"
 #include "nextpnr.h"
 #include "pins.h"
diff --git a/xilinx/pack_io_xc7.cc b/xilinx/pack_io_xc7.cc
index 3058cde0..29c75574 100644
--- a/xilinx/pack_io_xc7.cc
+++ b/xilinx/pack_io_xc7.cc
@@ -23,6 +23,7 @@
 #include <iterator>
 #include <queue>
 #include <unordered_set>
+#include <set>
 #include "cells.h"
 #include "chain_utils.h"
 #include "design_utils.h"
diff --git a/xilinx/pack_io_xcup.cc b/xilinx/pack_io_xcup.cc
index d59c55e0..421ba8ba 100644
--- a/xilinx/pack_io_xcup.cc
+++ b/xilinx/pack_io_xcup.cc
@@ -22,6 +22,7 @@
 #include <iterator>
 #include <queue>
 #include <unordered_set>
+#include <set>
 #include "cells.h"
 #include "chain_utils.h"
 #include "design_utils.h"
@@ -797,4 +798,4 @@ void USPacker::pack_idelayctrl()
     generic_xform(ioctrl_rules);
 }
 
-NEXTPNR_NAMESPACE_END
\ No newline at end of file
+NEXTPNR_NAMESPACE_END
